### PR TITLE
Closes softlayer/sl-ember-components#546

### LINKED ISSUE #546: "DEPRECATION: Ember.Handlebars.registerHelper is deprecated, please refactor to Ember.Helper.helper. [deprecation id: ember-htmlbars.handlebars-register-helper]
When accessing demo application

### PR DESCRIPTION
**Pull request by jonathandavidson**
In order to resolve deprecations for Ember.Handlebars.registerHelper the following was completed:

1. Removed `get-key` helper since it was no longer used anywhere in the add-on.
2. Removed `render-dynamic` helper in favor of the built-in `component` helper to follow best practice.
  * Removed instances of `render-dynamic` helper in `sl-grid` and `sl-tab-pane`.
  * Updated dummy app for `sl-grid` and `sl-tab-panel` to reflect new usage.
4. Removed register-helpers initializer since no helpers remain to be initialized.
3. Since all helpers were removed any related tests and removed references to helpers in README.md and the demo pages.

Closes #546, #615, #237, and #238

### CODE REVIEW

File: tests/dummy/app/templates/demos/sl-tab-panel.hbs [hunk 1]
```diff
@@ -21,8 +26,13 @@
     <div class="col-lg-6">
         <h6>Rendered Component</h6>
         {{#sl-tab-panel}}
-            {{sl-tab-pane label="One" name="one" templateName="demos/tabs/one"}}
-            {{sl-tab-pane label="Two" name="two" templateName="demos/tabs/two"}}
+            {{#sl-tab-pane label="One" name="one"}}
+                {{render "demos.tabs.one"}}
```
Comment (notmessenger) [inline tests/dummy/app/templates/demos/sl-tab-panel.hbs]: The 11th bullet point in the http://emberjs.com/blog/2015/05/24/another-two-oh-status-update.html#toc_removals list states that the `{{render}}` helper is being removed in 2.x, being superseded in most cases by components

File: tests/dummy/app/templates/demos/sl-tab-panel.hbs [hunk 2]
```diff
@@ -21,8 +26,13 @@
...
+            {{/sl-tab-pane}}
+
+            {{#sl-tab-pane label="Two" name="two"}}
+                {{render "demos.tabs.two"}}
```
Comment (notmessenger) [inline tests/dummy/app/templates/demos/sl-tab-panel.hbs]: The 11th bullet point in the http://emberjs.com/blog/2015/05/24/another-two-oh-status-update.html#toc_removals list states that the `{{render}}` helper is being removed in 2.x, being superseded in most cases by components

File: tests/dummy/app/controllers/demos/sl-tab-panel.js [hunk 3]
```diff
@@ -1,7 +1,5 @@
 import Ember from 'ember';
 
 export default Ember.Controller.extend({
-    needs: [ 'demos/tabs/one' ],
-
     one: Ember.computed.alias( 'controllers.demos/tabs/one' )
```
Comment (notmessenger) [inline tests/dummy/app/controllers/demos/sl-tab-panel.js]: This reference will break with removal of the `needs` won't it?

Comment (jonathandavidson) [inline tests/dummy/app/controllers/demos/sl-tab-panel.js]: Didn't break anything, but yes it absolutely needs to be removed.

File: tests/dummy/app/controllers/demos/sl-tab-panel.js [hunk 4]
```diff
@@ -1,7 +1,3 @@
 import Ember from 'ember';
 
-export default Ember.Controller.extend({
-    needs: [ 'demos/tabs/one' ],
-
-    one: Ember.computed.alias( 'controllers.demos/tabs/one' )
-});
+export default Ember.Controller.extend();
```
Comment (notmessenger) [inline tests/dummy/app/controllers/demos/sl-tab-panel.js]: Is this file even needed anymore?

**Pull request merged**